iothub/service/src/IoTHubSasCredential.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+#if !NET451
+
+using Azure;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Shared access signature credential used to authenticate with IoT hub.
+    /// </summary>
+    public class IoTHubSasCredential : AzureSasCredential
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="IoTHubSasCredential"/>.
+        /// </summary>
+        /// <param name="signature">Shared access signature used to authenticate with IoT hub.</param>
+        /// <param name="expiresOnUtc">The shared access signature expiry in UTC.</param>
+        public IoTHubSasCredential(string signature, DateTime expiresOnUtc) : base(signature)
+        {
+            ExpiresOnUtc = expiresOnUtc;
+        }
+
+        /// <summary>
+        /// Updates the shared access signature. This is intended to be used when you've
+        /// regenerated your shared access signature and want to update clients.
+        /// </summary>
+        /// <param name="signature">Shared access signature used to authenticate with IoT hub.</param>
+        /// <param name="expiresOnUtc">The shared access signature expiry in UTC.</param>
+        public void Update(string signature, DateTime expiresOnUtc)
+        {
+            Update(signature);
+            ExpiresOnUtc = expiresOnUtc;
+        }
+
+        /// <summary>
+        /// The shared access signature expiry in UTC.
+        /// </summary>
+        public DateTime ExpiresOnUtc { get; private set; }
```

Comment:
properties above methods

Reply:
Can you send me a link to where fields, methods, ctors should be? I can follow it when I write code.

Reply:
Is It as follows? And how do we order them based on public/private/protected?
Fields
Ctor
Properties
Methods

Reply:
Standards are here: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/doc/coding-style.md

Typical order is:
- fields (constants, static/readonly, static, other instance fields)
- ctor (public, internal protected, private)
- properties
- methods (public, internal, protected, private)